coq/coq.el

```diff
@@ -1203,21 +1213,45 @@ be called and no command will be sent to Coq."
      ;; If user issued a printing option then t printing.
      (and (string-match-p "\\(S\\|Uns\\)et\\s-+Printing" cmd)
           (> (length coq-last-but-one-proofstack) 0)))
-    (list "Show."))
+    (list "Show."
+    (when coq-show-proof-stepwise
+        (or
+    (when (eq coq-diffs 'off) "Show Proof.")
+    (when (eq coq-diffs 'on) "Show Proof Diffs.")
+    (when (eq coq-diffs 'removed) "Show Proof Diffs removed.")))))
+   
    ((or
      ;; If we go back in the buffer and the number of abort is less than
      ;; the number of nested goals, then Unset Silent and Show the goal
      (and (string-match-p "BackTo\\s-" cmd)
           (> (length coq-last-but-one-proofstack) coq--retract-naborts)))
     ;; "Set Diffs" always re-prints the proof context with (if enabled) diffs
-    (list "Unset Silent." (if (coq--post-v810) (coq-diffs) "Show.")))
+    (list "Unset Silent." (if (coq--post-v810) (coq-diffs) "Show.")
+                                  (when coq-show-proof-stepwise
+                                    (or
+    (when (eq coq-diffs 'off) "Show Proof.")
+    (when (eq coq-diffs 'on) "Show Proof Diffs.")
+    (when (eq coq-diffs 'removed) "Show Proof Diffs removed.")))))
+
    ((or
      ;; If we go back in the buffer and not in the above case, then only Unset
      ;; silent (there is no goal to show). Still, we need to "Set Diffs" again
      (string-match-p "BackTo\\s-" cmd))
     (if (coq--post-v810)
-        (list "Unset Silent." (coq-diffs))
-      (list "Unset Silent.")))))
+        (list "Unset Silent." (coq-diffs) )
+      (list "Unset Silent.")))
+   ((or
+     ;; If starting a proof, Show Proof if need be
+     (coq-goal-command-str-p cmd)
```

Comment:
FYI I adeed this line, as otherwise after doing `(setq coq-show-proof-stepwise t)` and starting a proof, the proof term was not shown after processing the first line of the theorem (`Lemma … : …`)

I've just seen that this function has the following comment:

https://github.com/ProofGeneral/PG/blob/71748af5bc0f813d4c6a42f303d2c2cec85b3c9a/coq/coq-syntax.el#L1027-L1030

so are you OK all the same @Matafou with using it just once? basically even if the `coq-goal-command-str-p` does not perfectly detects all goal starts (you may speak of a heuristic), IMHO it is needed here to have a fully complete feature #487 …

Also, @CyrilAnac in you next commit, can you reindent the whole function `coq-empty-action-list-command` ? (basically by selecting all the code of the function and typing <kbd>TAB</kbd>)

Reply:
Yes I will 👍